common: move identhandler to common module

mlohvynenko opened the pull request:
This patch moves identhandler interfaces to the common module, as these interfaces are needed by both iam and cm components.

MykolaSuperman reviewed (approved):
`Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>`

mykola-kobets-epam reviewed (approved):
`Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>`

mykola-kobets-epam reviewed (approved):
`Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>`

mlohvynenko committed:
common: tools: use assign to add split string item

This patch improves safety of the split function by using
emplace back and assign instead of a single push back, to
ensure split item size does not exceed the destination item size.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>

mlohvynenko committed:
common: identprovider: rename identhandler to identprovider

This patch renames identhandler to identprovider and moves the
interface to the common module.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>

mlohvynenko committed:
common: identprovider: add documentation

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
